src/Compiler/Compile.hs

```diff
@@ -642,31 +677,56 @@ resolveModule term flags currentDir modules mimp
 
     where
       name = impFullName mimp
+      
+      tryLoadFromCache :: FilePath -> FilePath -> FilePath -> IOErr (Maybe (Module, Modules))
+      tryLoadFromCache iface root stem 
```

Comment:
This method, loads from the `cachedModules`, subsequently calling `loadFromModule` to load all depedencies.

Reply:
loadFromIface and loadFromSource are adjusted accordingly, to defer to this method when the module is not already in loaded, and prior to actually loading from iface / source